### PR TITLE
Enable freeform answer submit when feedback is hidden

### PR DESCRIPTION
**Pull request by mtyaka**
cf. [MKCIN-3890](https://edx-wiki.atlassian.net/browse/MCKIN-3890)

*Test Instructions*

1. Create a new subunit containing one problem builder freeform question unit, and another arbitrary unit.
2. Enable `pb_hide_feedback_if_attempts_remain` in your `XBLOCK_SETTINGS`.
3. Go to the unit with the freeform question, fill it in, and submit.
4. Go to the next unit, then come back to the unit with the freeform question.
5. Feedback should be hidden, but the submit button should be enabled

This PR also contains a patch that makes MRQ behavior when `pb_mcq_hide_previous_answer` option is enable more consistent with the way MCQs behave. To test:

1. Create a new subunit containing one problem builder MRQ question unit, and another arbitrary unit.
2. Enable `pb_mcq_hide_previous_answer` in your `XBLOCK_SETTINGS`.
3. Go to the unit with the MCQ question, fill it in, and submit.
4. Go to the next unit, then come back to the unit with the MCQ question.
5. Feedback should be hidden, checkmarks next to the choices should NOT be visible, and previously selected choices should not be selected.

### CODE REVIEW

File: problem_builder/mentoring.py [hunk 1]
```diff
@@ -478,6 +478,10 @@ def next_step_url(self):
         """
         return '/jump_to_id/{}'.format(self.next_step)
 
+    @property
+    def hide_feedback(self):
+        return self.get_option("pb_hide_feedback_if_attempts_remain") and not self.max_attempts_reached
+
```
Comment (itsjeyd) [inline problem_builder/mentoring.py]: @mtyaka I like this refactoring :)

File: problem_builder/public/js/mentoring_with_steps.js [hunk 2]
```diff
@@ -302,7 +302,8 @@ function MentoringWithStepsBlock(runtime, element) {
             var step = steps[i];
             var mentoring = {
                 setContent: setContent,
-                publish_event: publishEvent
+                publish_event: publishEvent,
+                step_builder: true
```
Comment (itsjeyd) [inline problem_builder/public/js/mentoring_with_steps.js]: @mtyaka Since this is a Boolean, could we call it `is_step_builder`?

Comment (mtyaka) [inline problem_builder/public/js/mentoring_with_steps.js]: @itsjeyd Good idea, I renamed the variable.

File: problem_builder/public/js/mentoring_standard_view.js [hunk 3]
```diff
@@ -35,10 +35,13 @@ function MentoringStandardView(runtime, element, mentoring) {
             }
         }
 
+        // Data may have change, we have to re-validate.
+        validateXBlock();
+
```
Comment (itsjeyd) [inline problem_builder/public/js/mentoring_standard_view.js]: @mtyaka `change` ⟶ `changed`. Also, is this still necessary now that answer blocks are handling validation themselves?

**Comment by itsjeyd** [comment]
:+1: once newest comment is addressed and build is green.

File: problem_builder/public/js/mentoring_standard_view.js (hunk 3 above)
Comment (mtyaka) [inline problem_builder/public/js/mentoring_standard_view.js]: @itsjeyd I fixed the typo, thanks for noticing. This probably isn't necessary for answer blocks anymore (although it shouldn't do any harm), but I believe it still makes sense for other blocks.

**Comment by mtyaka** [comment]
Thanks @itsjeyd!

**Pull request merged**